drivers/sensor/lis2dw12/lis2dw12.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * STMicroelectronics LIS2DW12 driver header
+ *
+ * Copyright (c) 2018 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __ACCEL_LIS2DW12_H
+#define __ACCEL_LIS2DW12_H
+
+#include <gpio.h>
+#include <misc/util.h>
+#include <sensor.h>
+
+#define SYS_LOG_DOMAIN "LIS2DW1212"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SENSOR_LEVEL
+#include <logging/sys_log.h>
+
+/* COMMON DEFINE FOR ACCEL SENSOR */
+#define LIS2DW12_EN_BIT		0x01
+#define LIS2DW12_DIS_BIT		0x00
+#define LIS2DW12_OUT_LEN		6
+
+/* temperature sensor */
+#define LIS2DW12_OUT_TEMP_L_ADDR	0x0d
+
+/* Who Am I  */
+#define LIS2DW12_WHO_AM_I_REG		0x0f
+#define LIS2DW12_WHO_AM_I		0x44
+
+#define LIS2DW12_CTRL1_ADDR		0x20
+#define LIS2DW12_LOW_POWER_MASK	0x03
+#define LIS2DW12_POWER_MODE_MASK	0x0c
+#define LIS2DW12_LOW_POWER_MODE	0x00
+#define LIS2DW12_HP_MODE		0x04
+#define LIS2DW12_ODR_MASK		0xf0
+
+enum lis2dh_powermode {
+	LIS2DW12_LOW_POWER_M1,
+	LIS2DW12_LOW_POWER_M2,
+	LIS2DW12_LOW_POWER_M3,
+	LIS2DW12_LOW_POWER_M4,
+	LIS2DW12_HIGH_PERF
+};
+
+/* Acc data rate for Low Power mode */
+#define LIS2DW12_MAX_ODR		1600
+
+enum lis2dh_odr {
+	LIS2DW12_ODR_POWER_OFF_VAL,
+	LIS2DW12_ODR_1_6HZ_VAL,
+	LIS2DW12_ODR_12_5HZ_VAL,
+	LIS2DW12_ODR_25HZ_VAL,
+	LIS2DW12_ODR_50HZ_VAL,
+	LIS2DW12_ODR_100HZ_VAL,
+	LIS2DW12_ODR_200HZ_VAL,
+	LIS2DW12_ODR_400HZ_VAL,
+	LIS2DW12_ODR_800HZ_VAL,
+	LIS2DW12_ODR_1600HZ_VAL
+};
+
+#if defined(CONFIG_LIS2DW12_ODR_1_6)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_1_6HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_12_5)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_12_5HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_25)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_25HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_50)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_50HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_100) || \
+	defined(CONFIG_LIS2DW12_ODR_RUNTIME)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_100HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_200)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_200HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_400)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_400HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_800)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_800HZ_VAL
+#elif defined(CONFIG_LIS2DW12_ODR_1600)
+	#define LIS2DW12_DEFAULT_ODR	LIS2DW12_ODR_1600HZ_VAL
+#endif
+
+/* Return ODR reg value based on data rate set */
+#define LIS2DW12_ODR_TO_REG(_odr) \
+	((_odr <= 1) ? LIS2DW12_ODR_1_6HZ_VAL : \
+	 (_odr <= 12) ? LIS2DW12_ODR_12_5HZ_VAL : \
+	 ((31 - __builtin_clz(_odr / 25))) + 3)
```

Comment:
Need to wrap `_odr` with parenthesis inside the macro. There are few other macros in this file that  need the same change.